src/components/Envelope.vue

```diff
@@ -483,6 +475,27 @@ export default {
 				id: this.data.databaseId,
 			})
 		},
+		hasSeenAcl() {
+			if (!this.data.myAcls) {
+				return true
+			}
+			return this.data.myAcls.indexOf('s') !== -1
+		},
+		hasArchiveAcl() {
+			if (!this.data.myAcls) {
+				return true
+			} if (this.data.myAcls.indexOf('t') !== -1 || this.data.myAcls.indexOf('e') !== -1) {
+				return true
+			} else {
+				return this.data.myAcls.indexOf('i') !== -1
+			}
```

Comment:
The second check should be && instead of || because both ACLs are needed to deleted the source message.

The third check should be combined with the second check. Also the target/archive mailbox should be checked instead of `this.data`.

### Something like:
```
if (!this.data.myAcls) {
	return true
}

return this.data.myAcls.indexOf('t') !== -1 
	&& this.data.myAcls.indexOf('e') !== -1 
	&& this.archiveMailbox.myAcls.indexOf('i')
```

---

The same logic applies to moving messages as well. An archive operation is basically a move to a fixed, pre-configured mailbox.